snooty/postprocess.py

```diff
@@ -413,6 +416,94 @@ def get_include_argument(node: n.Directive) -> str:
+            include_filepath = (
+                self.project_config.root / self.project_config.source / include_slug
+            )
+
+            # Attempt to read the literally included file
+            try:
+                with open(include_filepath) as file:
+                    text = file.read()
+
+            except OSError as err:
+                self.diagnostics[filename].append(
+                    CannotOpenFile(include_filepath, err.strerror, node.span + (-1,))
+                )
+                return
+
+            lines = text.split("\n")
+
+            # Locate the start_after query
+            start_after = 0
+            if "start-after" in node.options:
+                start_after_text = node.options["start-after"]
+                assert isinstance(start_after_text, str)
+                start_after = next(
+                    (idx for idx, line in enumerate(lines) if start_after_text in line),
+                    -1,
+                )
+                if start_after < 0:
+                    self.diagnostics[filename].append(
+                        InvalidLiteralInclude(
+                            f'"{start_after_text}" not found in {include_filepath}',
+                            node.span + (-1,),
+                        )
+                    )
+                    return
+
+                # Only increment start_after if text is specified, to avoid capturing the start_after_text
+                start_after += 1
+
+            # ...now locate the end_before query
+            end_before = len(lines)
+            if "end-before" in node.options:
+                end_before_text = node.options["end-before"]
+                assert isinstance(end_before_text, str)
+                end_before = next(
+                    (
+                        idx
+                        for idx, line in enumerate(lines, start=start_after)
+                        if end_before_text in line
+                    ),
+                    -1,
+                )
+                if end_before < 0:
+                    self.diagnostics[filename].append(
+                        InvalidLiteralInclude(
+                            f'"{end_before_text}" not found in {include_filepath}',
+                            node.span + (-1,),
+                        )
+                    )
+                    return
+                end_before -= start_after
+
+            # Check that start_after_text precedes end_before_text
+            if start_after >= end_before:
+                self.diagnostics[filename].append(
+                    InvalidLiteralInclude(
+                        f'"{end_before_text}" precedes {start_after_text} in {include_filepath}',
+                        node.span + (-1,),
+                    )
+                )
+                return
+
+            lines = lines[start_after:end_before]
+
+            if "dedent" in node.options:
+                assert isinstance(node.options["dedent"], int)
```

Comment:
`dedent` can be an integer _or_ a flag according to the [rstspec.toml](https://github.com/mongodb/snooty-parser/blob/master/snooty/rstspec.toml#L118), so this assertion is invalid. We should also add some handling for the flag variant, in which the dedent amount is determined automatically (as was done in `PendingLiteralInclude`).